tests/ui/manual_option_folding.rs

```diff
@@ -0,0 +1,31 @@
+#![feature(option_get_or_insert_default)]
+#![allow(dead_code, clippy::unnecessary_lazy_evaluations, clippy::unnecessary_literal_unwrap)]
+#![warn(clippy::manual_option_folding)]
+
+fn main() {
+    let mut opt: Option<i32> = Some(42);
+    opt.get_or_insert_with(|| 21);
```

Comment:
`get_or_insert` returns `&mut T`. The lint has to check, that this value is not used. It would also be good to have tests for this